### PR TITLE
Fix `IsNull` pruning expression generation without null_count statistics

### PR DESCRIPTION
**Pull request by alamb**
(this is a one line code change PR, the rest is tests)

# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/3042

 # Rationale for this change
See https://github.com/apache/arrow-datafusion/issues/3042

# What changes are included in this PR?
1. Set the `nullability` annotation correctly
2. Add test coverage

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/physical_optimizer/pruning.rs [hunk 1]
```diff
@@ -639,7 +639,7 @@ fn build_is_null_column_expr(
         Expr::Column(ref col) => {
             let field = schema.field_with_name(&col.name).ok()?;
 
-            let null_count_field = &Field::new(field.name(), DataType::UInt64, false);
+            let null_count_field = &Field::new(field.name(), DataType::UInt64, true);
```
Comment (alamb) [inline datafusion/core/src/physical_optimizer/pruning.rs:642]: This is the fix

File: datafusion/core/src/physical_optimizer/pruning.rs [hunk 2]
```diff
@@ -1761,4 +1819,39 @@ mod tests {
...
+    fn prune_int32_is_null() {
+        let (schema, statistics) = int32_setup();
+
+        // Expression "i IS NULL" when there are no null statistics,
+        // should all be kept
+        let expected_ret = vec![true, true, true, true, true];
+
+        // i IS NULL, no null statistics
+        let expr = col("i").is_null();
+        let p = PruningPredicate::try_new(expr, schema.clone()).unwrap();
```
Comment (alamb) [inline datafusion/core/src/physical_optimizer/pruning.rs:1833]: prior to the fix, this line would panic

File: datafusion/core/src/physical_optimizer/pruning.rs [hunk 3]
```diff
@@ -1761,4 +1819,39 @@ mod tests {
...
+            vec![
+                Some(0), // no nulls (don't keep)
+                Some(1), // 1 null
+                None,    // unknown nulls
+                None, // unknown nulls (min/max are both null too, like no stats at all)
+                Some(0), // 0 nulls (max=null too which means no known max) (don't keep)
+            ],
+        );
+
+        let expected_ret = vec![false, true, true, true, false];
```
Comment (alamb) [inline datafusion/core/src/physical_optimizer/pruning.rs:1849]: This case simply didn't have coverage before that I could find

**Review by Dandandan — APPROVED** [review]
LGTM

**Comment by alamb** [comment]
Thanks for the review @Dandandan

**Pull request merged**